weather/module.py

```diff
@@ -18,6 +21,93 @@ class Weather(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 
+    def _get_useful_data(self, json, ctx, lang_preference):
+        """
+        example json: https://wttr.in/praha?lang=sk&format=j1
+        get useful data from json as list of individual days
+        """
+        weather = []
+        # get individual days to extract data
+        num_of_forecast_days = 3 # number of days to get forecast for (including current day, max is 3)
+        day_phases = { # dict for getting the data from json easier (when you don't wan't some phase of day comment it)
+            "Morning": 2,
+            "Day": 4,
+            # "Evening": 6,
+            "Night": 7
+        }
+        for i in range(num_of_forecast_days):
+            cur_day = json['weather'][i]
+            day_dict = {
+                "date": cur_day['date'],
+            }
+            cur_day = cur_day['hourly']
+            for k, v in day_phases.items():
+                if lang_preference != 'en':
+                    day_dict.update({_(ctx, f"{k}"): {
+                        "state": cur_day[v][f'lang_{lang_preference}'][0]['value'],
+                        "temp": cur_day[v]['tempC'],
+                        "feels_like": cur_day[v]['FeelsLikeC'],
+                        "wind_speed": cur_day[v]['windspeedKmph']
+                    }})
+                else:
+                    day_dict.update({_(ctx, f"{k}"): {
+                        "state": cur_day[v]['weatherDesc'][0]['value'],
+                        "temp": cur_day[v]['tempC'],
+                        "feels_like": cur_day[v]['FeelsLikeC'],
+                        "wind_speed": cur_day[v]['windspeedKmph']
+                    }})
+
+            weather.append(day_dict)
+        return weather
+
+    def _create_embeds(self, ctx, name, lang_preference):
+        """create embeds for scrollable embed"""
+        url = f"https://wttr.in/{name}?format=j1&lang={lang_preference}"
+        request = requests.get(url)
+        # check status code of request for failures (lazy way)
+        if request.status_code != 200:
+            # return error embed
+            return utils.discord.create_embed(
+                author=ctx.message.author,
+                title='An error occured while getting weather info.',
+                error=True,
+            )
+
+        # create day embeds
+        try:
+            days = self._get_useful_data(request.json(), ctx, lang_preference)
+        except JSONDecodeError:
+            return
+        embeds = []
+        for day in days:
+            embed = utils.discord.create_embed(
+                author=ctx.message.author,
+                title=_(ctx, "Weather forecast for {date}").format(date=day['date'])
+            )
+            for k, v in day.items():
```

Comment:
Same as previous - I think that k and v should be renamed to something more exact.